agent/sighandlers/termination_handler.go

```diff
@@ -75,10 +80,10 @@ func FinalSave(saver statemanager.Saver, taskEngine engine.TaskEngine) error {
 
 	stateSaved := make(chan error)
 	saveTimer := time.AfterFunc(finalSaveTimeout, func() {
-		stateSaved <- errors.New("Timed out trying to save to disk")
+		stateSaved <- errors.New("final save: timed out trying to save to disk")
 	})
 	go func() {
-		log.Debug("Saving state before shutting down")
+		seelog.Debug("Saving state before shutting down")
 		stateSaved <- saver.ForceSave()
```

Comment:
Excuse my not-Go-savviness... Should this have a trap/catch with a log so in the event that the state was not saved correctly... its at least logged?

Reply:
The `saver.ForceSave` sends its results on down the `stateSaved` channel and its up to the other side to determine what to do about that. However.. we could be better on how we log failures and its context.

Reply:
The log line is above, on line 83.  That function will get triggered on a timeout so we have a record that the save failed.